platform/net/zephyr/src/mender-http.c

```diff
@@ -257,6 +371,26 @@ mender_http_response_cb(struct http_response *response, enum http_final_call fin
     }
 }
 
+static void
+artifact_response_cb(struct http_response *response, MENDER_ARG_UNUSED enum http_final_call final_call, void *user_data) {
+
+    assert(NULL != response);
+    assert(NULL != user_data);
+
+    /* Retrieve request context */
+    mender_artifact_download_data_t *dl_data = user_data;
```

Comment:
We should cast the void pointer to `(mender_artifact_download_data_t *) user_data`

Reply:
That's done by the assignment.

Reply:
TIL! Thanks